### PR TITLE
Add `{id:}` to short translation source strings

### PR DESCRIPTION
**Pull request by sgk**
Short words may have multiple meanings, and such words should be translated into different words or sentences in different contexts. The `{id:...}` part in translation source strings will not appear in web pages even when the locale is English or when no translation is available. We can use `{id:...}` to identify the context.

### CODE REVIEW

File: pxtblocks/blocklyloader.ts [hunk 1]
```diff
@@ -405,12 +405,12 @@ namespace pxt.blocks {
 
         // Do not remove this comment.
         // These are used for category names.
-        // lf("Loops")
-        // lf("Logic")
-        // lf("Variables")
-        // lf("Lists")
-        // lf("Text")
-        // lf("Math")
+        // lf("{id:cat}Loops")
```
Comment (pelikhan) [inline pxtblocks/blocklyloader.ts]: cat or dog? :) Let's use ``{id:blocks category}``

File: pxtblocks/blocklyloader.ts [hunk 2]
```diff
@@ -676,7 +676,7 @@ namespace pxt.blocks {
     function initContextMenu() {
         // Translate the context menu for blocks.
         let msg: any = Blockly.Msg;
-        msg.DUPLICATE_BLOCK = lf("Duplicate");
+        msg.DUPLICATE_BLOCK = lf("Duplicate{id:block}");
```
Comment (pelikhan) [inline pxtblocks/blocklyloader.ts]: Let's keep the id in front ``{id:block}Duplicate"

File: pxtblocks/blocklyloader.ts (hunk 1 above)
Comment (sgk) [inline pxtblocks/blocklyloader.ts]: OK ;)

File: pxtblocks/blocklyloader.ts (hunk 2 above)
Comment (sgk) [inline pxtblocks/blocklyloader.ts]: OK, if you like it.

The reason I choose "duplicate block" was that it's natural word order of English.

**Comment by sgk** [comment]
@pelikhan Thank you for merging. Will the translation source strings be reflected to Crowdin soon? I'd like to start translation once the new strings available on Crowdin.

**Comment by pelikhan** [comment]
I just uploaded the new strings.json -- so you should see the new identifiers.

**Pull request merged**